`@types/react` version mismatch 16.x & 17.x: Types of property 'capture' are incompatible

https://opencrvsworkspace.slack.com/archives/C02LU432JGK/p1660637893727149

The dependencies we're using want to install both 16.x and 17.x versions of `@types/react`. This caused a minor typing issue with `TextInput.tsx` and `PasswordInput.tsx`.

```
Error: src/components/forms/PasswordInput.tsx(175,10): error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: { form?: string | undefined; slot?: string | undefined; style?: (React.CSSProperties & import("/home/runner/work/opencrvs-core/opencrvs-core/packages/components/node_modules/@types/react/index").CSSProperties) | undefined; ... 292 more ...; hideIcon?: ReactNode; } & { ...; } & { ...; }): ReactElement<...>', gave the following error.
    Type '{ type: "text" | "password"; error?: boolean | undefined; touched?: boolean | undefined; focusInput?: boolean | undefined; ignoreMediaQuery?: boolean | undefined; hideBorder?: boolean | undefined; ... 287 more ...; ref: RefObject<...>; }' is not assignable to type '{ form?: string | undefined; slot?: string | undefined; style?: (React.CSSProperties & React.CSSProperties) | undefined; ... 292 more ...; hideIcon?: ReactNode; }'.
      Types of property 'capture' are incompatible.
        Type 'string | boolean | undefined' is not assignable to type 'boolean | "user" | "environment" | undefined'.
          Type 'string' is not assignable to type 'boolean | "user" | "environment" | undefined'.
  Overload 2 of 2, '(props: StyledComponentPropsWithAs<"input", any, IPasswordInputProps, never, "input", "input">): ReactElement<StyledComponentPropsWithAs<"input", any, IPasswordInputProps, never, "input", "input">, string | ... 1 more ... | (new (props: any) => Component<...>)>', gave the following error.
    Type '{ type: "text" | "password"; error?: boolean | undefined; touched?: boolean | undefined; focusInput?: boolean | undefined; ignoreMediaQuery?: boolean | undefined; hideBorder?: boolean | undefined; ... 287 more ...; ref: RefObject<...>; }' is not assignable to type '{ form?: string | undefined; slot?: string | undefined; style?: (React.CSSProperties & React.CSSProperties) | undefined; ... 292 more ...; hideIcon?: ReactNode; }'.
      Types of property 'capture' are incompatible.
        Type 'string | boolean | undefined' is not assignable to type 'boolean | "user" | "environment" | undefined'.
          Type 'string' is not assignable to type 'boolean | "user" | "environment" | undefined'.
```

--- a/package.json
+++ b/package.json
@@ -62,6 +62,8 @@
     "//": "See https://github.com/facebook/create-react-app/issues/11773",
     "react-error-overlay": "6.0.9",
     "babel-loader": "8.1.0",
-    "underscore": ">1.12.1"
+    "underscore": ">1.12.1",
+    "//": "See https://github.com/opencrvs/opencrvs-core/issues/3820",
+    "@types/react": "^16.9.4"
   }
 }

